src/cli_driver.rs

```diff
@@ -449,14 +448,30 @@ impl ProcessingSession {
             };
 
             if let Some(par) = tex_path.parent() {
-                fs_root = par;
+                output_path = par;
                 io_builder.filesystem_root(par);
             } else {
                 return Err(ErrorKind::Msg(format!("can't figure out a parent directory for input path \"{}\"",
                                                   tex_path.to_string_lossy())).into());
             }
         }
 
+        if let Some(dir) = args.value_of("outdir") {
+            output_path = {
+                let tmp = Path::new(dir);
+                match tmp.extension() {
+                    Some(_) => {
+                        tt_warning!(status, "file extension in `-o/--outdir`; only the directory will be used");
+                        tmp.parent().unwrap()
+                        }
+                    None => &tmp,
+                }
+            };
+            if !output_path.exists() {
```

Comment:
On the other hand, I think here the check should use `is_dir`, not `exists`. I think this is closer to what we want and it will handle the case where the user inadvertently passes a filename instead of a directory name to the `-o` option.